### PR TITLE
Changelog

### PR DESCRIPTION
**Pull request by adsharma**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Unreleased
+
+### Added
+- Migrate API translation code to plugin infra
+- Support for self transpilation with improved error reporting
```
Comment (jayvdb) [inline CHANGELOG.md:5]: this is just a testing thing; has no outside benefit as self-transpilation renders junk.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+- New target: python. Transpiles untyped python to typed python.
+- Directory mode. Cross module type inference.
```
Comment (jayvdb) [inline CHANGELOG.md:7]: "Directory mode" previously existed.

Is there a test which does "Cross module type inference." ?  If they exist, there are not many of them, so I'd be wanting to create a few more tests to verify it works reasonably well.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+- Experimental: pyO3 extension for rust
```
Comment (jayvdb) [inline CHANGELOG.md:8]: Perhaps pyO3 mode could be converted into a new target "pyO3" or "rust+pyO3" ?

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+- Rust: python sealed classes -> Rust Enum
```
Comment (jayvdb) [inline CHANGELOG.md:9]: i think we need to hyperlink sealed to adt repo.

File: CHANGELOG.md (hunk 2 above)
Comment (adsharma) [reply CHANGELOG.md:7]: Opened #391

File: CHANGELOG.md (hunk 1 above)
Comment (adsharma) [reply CHANGELOG.md:5]: Will remove

File: CHANGELOG.md (hunk 3 above)
Comment (adsharma) [reply CHANGELOG.md:8]: It's a sub-target of rust, not a standalone one. A large module with a small interface to python would be mostly the same as rust. Only the code that interfaces with python would be transpiled differently.

**Comment by jayvdb** [comment]
https://github.com/adsharma/py2many/issues/149#issue-873158088 said we had "79% code coverage", but https://github.com/adsharma/py2many/runs/2571696956#step:12:518 says 84%

last `main` commit says 90% https://github.com/adsharma/py2many/runs/3005235149#step:16:748 `\*/` . That is the magic threshold for me.  And reaching 90% means we dont even need to say what the prior coverage level was, as anyone who cares about this line in the changelog will know each percentage point in that region is hard work.

And we should now start requiring 90%.  Note that if we integrate SonarCloud (https://github.com/adsharma/py2many/issues/347), we can make that a soft requirement defined in the "Quality Gate", and it will fail the PR, but that can be ignored when merging to `main` (and ideally then followed by another PR which boosts the coverage back over the threshold).

**Comment by jayvdb** [comment]
new test cases are IMO all new working functionality worth putting in changelog.  fib_with_argparse/sealed/with_open are all Rust-only, and IMO Rust should have a separate sub-section in the relnotes given it usually gets features first.

```
❯ git diff --name-only --diff-filter=A v0.2.1..HEAD -- tests/cases
      1 tests/cases/bitops.py
      2 tests/cases/built_ins.py
      3 tests/cases/byte_literals.py
      4 tests/cases/comment_unsupported.py
      5 tests/cases/dict.py
      6 tests/cases/fib_with_argparse.py
      7 tests/cases/sealed.py
      8 tests/cases/sys_argv.py
      9 tests/cases/sys_exit.py
     10 tests/cases/with_open.py
```

Test successes since the last release
```
❯ GIT_PAGER=cat git diff --name-only --diff-filter=A v0.2.1..HEAD -- tests/expected | \grep --color=always -v '\.py' | cut -d '.' -f 1 | uniq -c | sort -rn
   7 tests/expected/sys_exit  # all langs
   7 tests/expected/sys_argv
   7 tests/expected/comment_unsupported
   7 tests/expected/built_ins  # min() was fixed
   7 tests/expected/bitops
   4 tests/expected/nested_dict  # rust & julia & go & kotlin
   3 tests/expected/dict # rust & julia & dart
   2 tests/expected/byte_literals # rust & julia
   1 tests/expected/with_open  # rust
   1 tests/expected/sealed  # rust
   1 tests/expected/fib_with_argparse  # rust
```

**Pull request merged**